contracts/token/BiconomyTokenPaymaster.sol

```diff
@@ -101,28 +95,28 @@ contract BiconomyTokenPaymaster is BasePaymaster, ReentrancyGuard, TokenPaymaste
     /**
      * Designed to enable tracking how much fees were charged from the sender and in which ERC20 token
      * More information can be emitted like exchangeRate used, what was the source of exchangeRate etc*/
-    event TokenPaymasterOperation(address indexed sender, address indexed token, uint256 indexed totalCharge, uint256 premium, bytes32 userOpHash, uint256 exchangeRate, ExchangeRateSource priceSource);
+    event TokenPaymasterOperation(address indexed sender, address indexed token, uint256 indexed totalCharge, address oracleAggregator, uint256 premium, bytes32 userOpHash, uint256 exchangeRate, ExchangeRateSource priceSource);
 
     /**
      * Notify in case paymaster failed to withdraw tokens from sender
      */
     event TokenPaymentDue(address indexed token, address indexed account, uint256 indexed charge);
 
+    /**
+     * Record the information of swap made on dex router and native tokens deposited
+     */
+    event TokenSwappedAndGasDeposited(address indexed dexRouterAddress, address indexed token, bytes indexed routerCalldata, bool success, uint256 gasAmountDeposited);
```

Comment:
Keeping bytes type field in the middle of the event might cause parsing issue on client side. Not sure about this but keeping bytes type field at the end might make it easy to parse the events data.

It if is indexed, then it might not cause this issue. 

Parsing of this event offchain should be verified once.

Reply:
it's indexed.